From eb67bd3417dd40b1efe7457d48f3d4c922cf2135 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 13 Oct 2021 12:55:29 +0300
Subject: [PATCH] [SW-59081] gaudi2: implement pcie skip check

For gaudi2, dump_sel[4] bit indicates whether to skip or perform
pcie init. This patch checks the bit as part of BOOT_STRAP_L
register and returns values accordingly.

Change-Id: I29237bd2e13a20ee9379f5bae066a6a75a9688ee
---

diff --git a/arch/gaudi2/arc/asic.c b/arch/gaudi2/arc/asic.c
index d559b1f..cc2d753 100644
--- a/arch/gaudi2/arc/asic.c
+++ b/arch/gaudi2/arc/asic.c
@@ -27,12 +27,6 @@
 	return (base & ~SDRAM_BASE) | SDRAM_BASE_FULL;
 }
 
-u32 hl_skip_pcie_init(void)
-{
-	/* TODO: Use DumpSel[*] to skip PCIe initialization */
-	return 1;
-}
-
 /*
  * FW_SRAM layout:
  * ---------------
diff --git a/arch/gaudi2/arm/asic.c b/arch/gaudi2/arm/asic.c
index e2d9783..f27aa34 100644
--- a/arch/gaudi2/arm/asic.c
+++ b/arch/gaudi2/arm/asic.c
@@ -22,12 +22,6 @@
 	return ((base & ~SDRAM_BASE) | SDRAM_BASE_FULL);
 }
 
-u32 hl_skip_pcie_init(void)
-{
-	/* TODO: Use DumpSel[*] to skip PCIe initialization */
-	return 0;
-}
-
 bool ppboot_is_valid_img_desc(struct boot_desc *desc)
 {
 	u8 hash[SHA384_SUM_LEN];
diff --git a/arch/gaudi2/common/asic.c b/arch/gaudi2/common/asic.c
index 61c620c..72c0974 100644
--- a/arch/gaudi2/common/asic.c
+++ b/arch/gaudi2/common/asic.c
@@ -177,3 +177,9 @@
 	outbound_region.size = HOST_PHYS_SIZE_0;
 	hl_pci_set_outbound_region(&outbound_region);
 }
+
+
+u32 hl_skip_pcie_init(void)
+{
+	return hl_is_debug_mode(HL_DEBUG_HW_PCIE);
+}
